pkg/oauthserver/setup_test.go

```diff
@@ -0,0 +1,119 @@
+package oauthserver_test
+
+import (
+	"net/http"
+
+	oauthv1 "github.com/openshift/api/oauth/v1"
+	osinv1 "github.com/openshift/api/osin/v1"
+	fakeoauthclient "github.com/openshift/client-go/oauth/clientset/versioned/fake"
+	typedv1 "github.com/openshift/client-go/oauth/clientset/versioned/typed/oauth/v1"
+	bootstrap "github.com/openshift/library-go/pkg/authentication/bootstrapauthenticator"
+	"github.com/openshift/oauth-server/pkg/config"
+	"github.com/openshift/oauth-server/pkg/oauthserver"
+	"github.com/openshift/oauth-server/pkg/userregistry/identitymapper"
+	"golang.org/x/crypto/bcrypt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	fakekube "k8s.io/client-go/kubernetes/fake"
+)
+
+const (
+	// testUser == kubeadmin
+	testPassword = "testpasswordxtestpasswordx" // must have at least len 23
+
+	clientKey   = "client_id"
+	clientValue = "my_client"
+
+	responseTypeKey   = "response_type"
+	responseTypeValue = "code"
+)
+
+type BootstrapUserIDP func() (*bootstrap.BootstrapUserData, bool, error)
```

Comment:
Unexport, rename to `mockBootstrapUser`

Reply:
wow, the naming is strong with you :) super helpful thx.

Reply:
done